pkg/logs/internal/tailers/file/tailer.go

```diff
@@ -126,6 +127,18 @@ type TailerOptions struct {
 	Rotated       bool                  // Optional
 }
 
+type DataPoint struct {
+	Timestamp int64
+	Value     int64
+}
+
+type MovingSum struct {
+	DataPoints   []*DataPoint
+	TimeWindow   int64
+	timeProvider func() int64
```

Comment:
We've been using `github.com/benbjohnson/clock` as a clock abstraction elsewhere,  For example, in the [serializer](https://github.com/datadog/datadog-agent/blob/a196162164ef96e758006120fcfa134a90f2341a/pkg/serializer/serializer.go#L131-L132) or in the [collector](https://github.com/datadog/datadog-agent/blob/a196162164ef96e758006120fcfa134a90f2341a/pkg/collector/worker/utilization_tracker.go#L50) packages. Would it make sense to follow the pattern here?